src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(number) {
+      this.result += number;
+    },
+    subtract(number) {
+      this.result -= number;
+    },
+    multiply(number) {
+      this.result *= number;
+    },
+    divide(number) {
+      if (number !== 0) {
+        this.result /= number;
+      } else {
+        Error('Cannot divide by zero');
+      }
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
+      } else {
+        Error('Provided callback is not a function');
```

Comment:
Similar to the previous comment, you should use `throw` with the `Error` function to raise an error when the callback is not a function.